htmap/maps.py

```diff
@@ -142,6 +142,74 @@ def load(cls, tag: str) -> "Map":
 
             return cls(tag=tag, map_dir=map_dir,)
 
+    def _repr_html_(self):
+
+        table = self._repr_html_table()
+        grid = ""
+
+        both = [
+            "<table>",
+            "<tr>",
+            "<td>",
+            table,
+            "</td>",
+            "<td>",
+            grid,
+            "</td>",
+            "</tr>",
+            "</table>",
+        ]
+        return "\n".join(both)
+
+    def _repr_header_(self):
+        return "<th> TAG </th><th> HELD </th><th> ERRORED </th><th> IDLE </th><th> RUNNING </th><th> COMPLETED </th><th> Local Data </th><th> Max Memory </th><th> Max Runtime </th><th> Total Runtime </th>"
+
+    def _repr_grid_(self):
+        sc = collections.Counter(self.component_statuses)
+        row: Dict[str, Union[str, int, float]] = {"tag": self.tag}
+
+        for status in state.ComponentStatus.display_statuses():
+            row[status.value.lower()] = sc[status]
+
+        tag = row["tag"]
+        held = row["held"]
+        errored = row["errored"]
+        idle = row["idle"]
+        running = row["running"]
+        completed = row["completed"]
+
+        local_data = utils.num_bytes_to_str(self.local_data)
+        max_memory = utils.num_bytes_to_str(max(self.memory_usage) * 1024 * 1024)
+        max_runtime = str(max(self.runtime))
+        total_runtime = str(sum(self.runtime, datetime.timedelta()))
```

Comment:
The `row` dictionary seems unhelpful here; just pull the data directly from `self.component_statuses` (you can still use `state.ComponentStatus.display_statuses()` to get the ones to show), and use `self.tag` for the tag.